smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -52,15 +52,24 @@ contract MixinPurchase is
       // Assign a new tokenId (if a new owner or previously transferred)
       _assignNewTokenId(toKey);
       _recordOwner(_recipient, toKey.tokenId);
+
+      // trigger event
+      emit Transfer(
+        address(0), // This is a creation.
+        _recipient,
+        toKey.tokenId
+      );
     }
 
     if (toKey.expirationTimestamp >= block.timestamp) {
       // This is an existing owner trying to extend their key
       toKey.expirationTimestamp = toKey.expirationTimestamp.add(expirationDuration);
+      emit ExpirationChanged(toKey.tokenId, expirationDuration, true);
     } else {
       // SafeAdd is not required here since expirationDuration is capped to a tiny value
       // (relative to the size of a uint)
       toKey.expirationTimestamp = block.timestamp + expirationDuration;
+      emit ExpirationChanged(toKey.tokenId, expirationDuration, true);
```

Comment:
Hmm here we technically added more than just expirationDuration... 

I'm not sure this matters though :)  We might not need this level of detail - maybe in another PR we could change the event to `ExpirationChanged(tokenId, newExpiration)`.

Hmm another thought on that, using the user's address in the event is probably more useful than tokenId given that each user can only have one id.  That would allow front end implementations to listen for events filtered by the user's address.  So maybe `ExpirationChanged(address user, uint newExpiration)`

Reply:
>we technically added more than just expirationDuration

Hmm. Yeah, I should probably change this to be using `block.timestamp + expirationDuration` as the amount in the event. I was trying to reuse the existing event `ExpirationChanged` here, which I originally created for use in the `timeMachine` function. The purpose there was/is to log the amount of time added or removed to a given key. It seems like it would not achieve this if we logged the `newExpiration` there instead... Logging the owner's address instead of the tokenId could work. It makes me reconsider trying to reuse this event vs creating a new one specifically tailored to key renewals...

Reply:
It makes me reconsider trying to reuse this event vs creating a new one specifically tailored to key renewals, as the semantics are a bit different (logging the amount of time added or removed from a key, vs logging the updated expiration. I realize that one can be derived from the other, so I'm not feeling too strongly either way about reusing the existing event vs creating a new one.

Reply:
A new one for this use case seems good.  Maybe `RenewKeyPurchase(address indexed owner, uint newExpiration)`

Reply:
or maybe to be closer to the purchase event: `RenewKeyPurchase(address indexed owner, uint indexed tokenId)`.  ... or maybe both tokenId and newExpiration.

hmm event best practices need some work :)

Reply:
Ok. I think this is probably a better way forward too. I'll make the changes and push one more time. I don't want to be guessing about the meaning of values in events, or introduce more event-related bugs down the road.